cpp/dolfinx/fem/assemble_scalar_impl.h

```diff
@@ -228,6 +265,25 @@ T assemble_scalar(
         perms);
   }
 
+  for (int i : M.integral_ids(IntegralType::vertex))
+  {
+    auto fn = M.kernel(IntegralType::vertex, i, 0);
+    assert(fn);
+
+    auto& [coeffs, cstride] = coefficients.at({IntegralType::vertex, i});
+
+    std::span<const std::int32_t> data = M.domain(IntegralType::vertex, i, 0);
+    assert(data.size() * cstride == coeffs.size());
+
+    md::mdspan<const std::int32_t,
+               md::extents<std::size_t, md::dynamic_extent, 2>>
+        cell_and_vertex(data.data(), data.size() / 2, 2);
```

Comment:
If we want to be consistent with the other integral types, we should remove the temporary variable declaration cell_and_vertex, and just do the casting inside `assemble_vertices` below.